src/main/java/core/basesyntax/service/impl/CsvFileWriterServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.CsvFileWriterService;
+import core.basesyntax.service.ReportGenerationService;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class CsvFileWriterServiceImpl implements CsvFileWriterService {
+
+    @Override
+    public void writeToFile(String path, ReportGenerationService reportGenerationService) {
```

Comment:
you should pass the result of report generation to this method instead of ReportGenerationService itself